osu.Game/Rulesets/BindableRulesetSelector.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+
+namespace osu.Game.Rulesets
+{
+    public abstract class BindableRulesetSelector : RulesetSelector
+    {
+        [BackgroundDependencyLoader]
+        private void load(Bindable<RulesetInfo> parentRuleset)
+        {
+            Current.BindTo(parentRuleset);
+            Current.BindValueChanged(OnRulesetChanged);
+        }
+
+        protected abstract void OnRulesetChanged(ValueChangedEvent<RulesetInfo> e);
```

Comment:
Could just use `Current.BindValueChanged()` instead. Kinda fits in-line with my above comment.